src/Microsoft.TemplateEngine.Abstractions/TemplateFiltering/ITemplateMatchInfo.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Abstractions.TemplateFiltering
+{
+    /// <summary>
+    /// Template with information about matching the filters.
+    /// </summary>
+    public interface ITemplateMatchInfo
+    {
+        /// <summary>
+        /// True when the template matched all the filters applied.
+        /// </summary>
+        bool IsMatch { get; }
+
+        /// <summary>
+        /// True when the template matched at least one of the filters applied.
+        /// </summary>
+        bool IsPartialMatch { get; }
```

Comment:
I think we should drop `IsMatch` and `IsPartialMatch` properties and move logic from `TemplateMatchInfo` that implements this 2 properties into `ExactCriteria` and `PartialCriteria` respectively.

Reply:
I think these properties might be useful but names are confusing with MatchKind.Exact and MatchKind.Partial.
What if we rename it to: `MatchesAllFilters` and `MatchesOneFilter`?

Reply:
I still think that this properties are "useless" or `ExactCriteria ` and `PartialCriteria` is "useless"...

I would prefer that if job of `ExactCriteria `/`PartialCriteria` is to do the filtering by some logic, it should do it on top of `MatchDisposition` so its same for all `ITemplateMatchInfo` implementations, each implementation of `ITemplateMatchInfo` will need to reimplement same logic for this properties... Seems a bit wasteful...

So ideally in my opinion... drop this 2 properties and rename `ExactCriteria`=>`MathesAllCriteria` and `PartialCriteria`=>`MatchesAtLeastOneCriteria`